workflows/prognostic_c48_run/prepare_config.py

```diff
@@ -100,11 +101,17 @@ def user_config_from_dict_and_args(config_dict: dict, args) -> UserConfig:
         config_dict.get("namelist", {}).get("fv_core_nml", {}).get("nudge", False)
     )
 
-    prephysics: Optional[MachineLearningConfig]
-    if "prephysics" in config_dict:
+    prephysics: Optional[Union[MachineLearningConfig, PrescriberConfig]]
```

Comment:
hopefully we can replace this whole function with `dacite.from_dict(UserConfig, config_dict)` soon which will remove the necessity for all fo this.